kornia/augmentation/_2d/intensity/random_channel_dropout.py

```diff
@@ -0,0 +1,95 @@
+        keepdim: whether to keep the output shape the same as input (True) or broadcast it
+                 to the batch form (False).
+
+    Shape:
+        - Input: :math:`(C, H, W)` or :math:`(B, C, H, W)`
+        - Output: :math:`(B, C, H, W)` or :math:`(C, H, W)`
+
+    Examples:
+        >>> import torch
+        >>> torch.random.initial_seed(1)
+        >>> img = torch.ones(1, 3, 5, 5)
+        >>> out = RandomChannelDropout(p=1)(img)
+        >>> out.shape
+        torch.Size([1, 3, 5, 5])
+        >>> out
+        tensor([[[[False, False, False, False, False],
+          [False, False, False, False, False],
+          [False, False, False, False, False],
+          [False, False, False, False, False],
+          [False, False, False, False, False]],
+
+         [[ True,  True,  True,  True,  True],
+          [ True,  True,  True,  True,  True],
+          [ True,  True,  True,  True,  True],
+          [ True,  True,  True,  True,  True],
+          [ True,  True,  True,  True,  True]],
+
+         [[ True,  True,  True,  True,  True],
+          [ True,  True,  True,  True,  True],
+          [ True,  True,  True,  True,  True],
+          [ True,  True,  True,  True,  True],
+          [ True,  True,  True,  True,  True]]]])
+    """
+
+    def __init__(
+        self,
+        channel_drop_range: Tuple[int, int] = (1, 1),
+        fill_value: Union[int, float] = 0,
+        same_on_batch: bool = False,
+        p: float = 0.5,
+        keepdim: bool = False,
+    ) -> None:
+        super().__init__(p=p, same_on_batch=same_on_batch, p_batch=1.0, keepdim=keepdim)
+        self.channel_drop_range = channel_drop_range
+        self.fill_value = fill_value
+        self.min_channels = channel_drop_range[0]
+        self.max_channels = channel_drop_range[1]
+        KORNIA_CHECK(
+            1 <= self.min_channels <= self.max_channels, f"Invalid channel_drop_range. Got: {channel_drop_range}"
+        )
+
+    def apply_transform(
+        self, input: Tensor, params: Dict[str, Tensor], flags: Dict[str, Any], transform: Optional[Tensor] = None
+    ) -> Tensor:
+        KORNIA_CHECK(
+            not (
+                (len(input.shape) == 3 and input.shape[0] == 1)
+                or (len(input.shape) == 4 and input.shape[1] == 1)
+                or (len(input.shape) == 2)
+            ),
+            "Only one channel. ChannelDropout is not defined.",
+        )
+        if len(input.shape) == 3:
+            num_channels = input.shape[0]
+        else:
+            num_channels = input.shape[1]
+        KORNIA_CHECK(self.max_channels < num_channels, "Can not drop all channels in ChannelDropout.")
+        num_drop_channels = randint(self.min_channels, self.max_channels)
+
+        channels_to_drop = sample(range(num_channels), k=num_drop_channels)
+
+        input_cp = clone(input)
+        if len(input.shape) == 3:
+            input_cp[channels_to_drop, ...] = self.fill_value
```

Comment:
```suggestion
            input_cp[..., channels_to_drop, :, :] = self.fill_value
```
not clear but you could do something like